Add Fivetran guide (#1291)
neondatabase/website: content/docs/guides/logical-replication-fivetran.md

---
title: Replicate data with Fivetran
subtitle: Learn how to replicate data from Neon with Fivetran
enableTableOfContents: true
isDraft: false
updatedOn: '2023-12-21T15:11:12.212Z'
---

Neon's logical replication feature allows you to replicate data from your Neon Postgres database to external destinations. 

> [Fivetran](https://fivetran.com/) is an automated data movement platform that helps you centralize data from disparate sources, which you can manage directly from your browser. Fivetran extracts your data and loads it into your data destination.

Reviewer: ```suggestion
[Fivetran](https://fivetran.com/) is an automated data movement platform that helps you centralize data from disparate sources, which you can then manage directly from your browser. Fivetran extracts your data and loads it into your data destination.
```

In this guide, you will learn how to define a Neon Postgres database as a data source in Fivetran so that you can replicate data to one or more of Fivetran's supported destinations.

## Prerequisites

- A [Fivetran account](https://fivetran.com/)
- A [Neon account](https://console.neon.tech/)

## Enable logical replication in Neon

<Admonition type="important">

> Enabling logical replication modifies the Postgres `wal_level` configuration parameter, changing it from `replica` to `logical` for all databases in your Neon project. Once the `wal_level` setting is changed to `logical`, it cannot be reverted. Enabling logical replication also restarts all computes in your Neon project, meaning active connections will be dropped and have to reconnect.

Reviewer: ```suggestion
Enabling logical replication modifies the Postgres `wal_level` configuration parameter, changing it from `replica` to `logical` for all databases in your Neon project. Once the `wal_level` setting is changed to `logical`, it cannot be reverted. Enabling logical replication also restarts all computes in your Neon project, meaning active connections will be temporarily dropped before automatically reconnecting.
```

Reviewer: To avoid any implication that users will have to manually reconnect.

</Admonition>

To enable logical replication in Neon:

1. Select your project in the Neon console.
2. On the Neon **Dashboard**, select **Settings**.
3. Select **Beta**.
4. Click **Enable** to enable logical replication.

You can verify that logical replication is enabled by running the following query from the [Neon SQL Editor](/docs/get-started-with-neon/query-with-neon-sql-editor):

```sql
SHOW wal_level;
 wal_level 
-----------
 logical
```

## Create a Postgres role for replication

> It is recommended that you create a dedicated Postgres role for replicating data. The role must have the `REPLICATION` privilege. The default Postgres role created with your Neon project and roles created using the Neon Console, CLI, or API are granted membership in the [neon_superuser](/docs/manage/roles#the-neonsuperuser-role) role, which has the required `REPLICATION` privilege.

Reviewer: Is this a general Postgres recommendation, or Neon? If Neon, consider:
"Neon recommends..." or "We recommend..."
I'd probably go with "We recommend..." regardless actually. In favor of the move towards "opinionated" advice.

Author: changed as suggested

<Tabs labels={["Neon Console", "CLI", "API"]}>

<TabItem>

To create a role in the Neon Console:

1. Navigate to the [Neon Console](https://console.neon.tech).
2. Select a project.
3. Select **Roles**.
4. Select the branch where you want to create the role.
4. Click **New Role**.
5. In the role creation dialog, specify a role name.

> 6. Click **Create**. The role is created and you are provided with the password for the role.

Reviewer: ```suggestion
6. Click **Create**. 

The role is now created and you are provided with the password for the role, which you can show, copy, or download.
```

Fivetran uses the `pgoutput` plugin in Postgres for decoding WAL changes into a logical replication stream. To create a replication slot called `fivetran_slot` that uses the `pgoutput` plugin, run the following command on your database using your replication role:

```sql
SELECT pg_create_logical_replication_slot('fivetran_pgoutput_slot', 'pgoutput');
```

The name assigned to the replication slot is `fivetran_pgoutput_slot`. You will need to provide this name when you set up your Fivetran source.

## Create a Postgres source in Fivetran

1. Log in to your [Fivetran](https://fivetran.com/) account.
1. On the **Select your datasource** page, search for the **PostgreSQL** source and click **Set up**.
1. In your connector setup form, enter a value for **Destination Schema Prefix**. This prefix applies to each replicated schema and cannot be changed once your connector is created. In this example, we'll use `neon` as the prefix.
1. Enter the connection details for your Neon database. You can get these details from your Neon connection string, which you'll find in the **Connection Details** widget on the **Dashboard** of your Neon project. 

>     For example, given a connection string like this:

Reviewer: ```suggestion
    For example, let's say this is your connection string:
```

    <CodeBlock shouldWrap>

    ```bash
    postgres://alex:AbC123dEf@ep-cool-darkness-123456.us-east-2.aws.neon.tech/dbname?sslmode=require
    ```

    </CodeBlock>

>     Enter the details in the Fivetran **Create a source** dialog as shown below. Your values will differ, with the exception of the port number.

Reviewer: ```suggestion
    From this string, enter the details in the Fivetran **Create a source** dialog as shown below. Your actual alues will differ, with the exception of the port number.
```
